stepflow-rs/crates/stepflow-load-balancer/src/main.rs

```diff
@@ -102,11 +102,32 @@ impl ProxyHttp for StepflowLoadBalancer {
 
     async fn upstream_request_filter(
         &self,
-        _session: &mut Session,
+        session: &mut Session,
         upstream_request: &mut RequestHeader,
         _ctx: &mut Self::CTX,
     ) -> PingoraResult<()> {
-        // Add headers for debugging
+        // Propagate W3C trace context headers for distributed tracing
```

Comment:
Would it make sense to instead propagate *all* headers (possibly removing one or two that are LB specific)?

We could basically pass through all headers, and then only drop `STEPFLOW_LB_*` or something if we have headers that the LB should terminate.

Reply:
Let's add this to the issue list for next week, but yes, i think propogating all headers (or terminating business case specific ones in the LB runtime) is ideal